src/commands/createWebApp/setPostPromptDefaults.ts

```diff
@@ -39,48 +39,74 @@ export async function setPostPromptDefaults(wizardContext: IWebAppWizardContext,
+                    return plan.name && plan.name.includes(defaultPlanName) && getResourceGroupFromId(nonNullProp(plan, 'id')).includes(defaultGroupName);
+                });
+
+                // when using appServicePlans.list, the numOfSites are all set to 0 so individually get each plan and look for one with less than 3 sites
+                for (const plan of defaultPlans) {
+                    if (plan.name) {
+                        const groupName: string = getResourceGroupFromId(nonNullProp(plan, 'id'));
+                        const fullPlanData: WebSiteManagementModels.AppServicePlan | undefined = await tryGetAppServicePlan(client, groupName, plan.name);
+                        if (fullPlanData && matchesTier(fullPlanData, newSkuTier) && !checkPlanForPerformanceDrop(fullPlanData)) {
+                            wizardContext.newResourceGroupName = groupName;
+                            wizardContext.newPlanName = plan.name;
+                            break;
+                        }
                     }
                 }
-            }
 
-            wizardContext.newResourceGroupName = wizardContext.newResourceGroupName || await siteNameStep.getRelatedName(wizardContext, defaultGroupName);
-            if (!wizardContext.newResourceGroupName) {
-                throw new Error(localize('noUniqueNameRg', 'Failed to generate unique name for resources. Use advanced creation to manually enter resource names.'));
-            }
+                wizardContext.newResourceGroupName = wizardContext.newResourceGroupName || await siteNameStep.getRelatedName(wizardContext, defaultGroupName);
+                if (!wizardContext.newResourceGroupName) {
+                    throw new Error(localize('noUniqueNameRg', 'Failed to generate unique name for resources. Use advanced creation to manually enter resource names.'));
+                }
 
-            wizardContext.newPlanName = await siteNameStep.getRelatedName(wizardContext, defaultPlanName);
-            if (!wizardContext.newPlanName) {
-                throw new Error(localize('noUniqnueNameAsp', 'Failed to generate unique name for app service plan. Use advanced creation to manually enter plan names.'));
+                wizardContext.newPlanName = await siteNameStep.getRelatedName(wizardContext, defaultPlanName);
+                if (!wizardContext.newPlanName) {
+                    throw new Error(localize('noUniqnueNameAsp', 'Failed to generate unique name for app service plan. Use advanced creation to manually enter plan names.'));
+                }
             }
+        } else {
+            wizardContext.newResourceGroupName = defaultGroupName;
+            wizardContext.newPlanName = defaultPlanName;
+        }
+    } catch (e) {
+        if (parseError(e).errorType === 'AuthorizationFailed') {
+            wizardContext.newResourceGroupName = defaultGroupName;
+            wizardContext.newPlanName = defaultPlanName;
```

Comment:
These changes are needed to catch this error and set these values to some arbitrary values so that they are not undefined and continue with the rest of the code as is. 
Undefined values for `newResourceGroupName` and `newPlanName` is not expected with ResourceGroupCreateStep, which will throw an error.